fenix/app/src/test/java/org/mozilla/fenix/PrivateBrowsingBindingTest.kt

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix
+
+import android.view.Window
+import android.view.WindowManager
+import kotlinx.coroutines.test.runTest
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.libstate.ext.waitUntilIdle
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.rule.MainCoroutineRule
+import mozilla.components.support.test.rule.runTestOnMain
+import mozilla.components.support.test.whenever
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.anyInt
+import org.mockito.Mockito.never
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.theme.ThemeManager
+import org.mozilla.fenix.utils.Settings
+
+class PrivateBrowsingBindingTest {
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+
+    private lateinit var themeManager: ThemeManager
+    private lateinit var window: Window
+    private lateinit var settings: Settings
+    private val retrieveWindow = { window }
+
+    private lateinit var appStore: AppStore
+
+    private lateinit var binding: PrivateBrowsingBinding
+
+    @Before
+    fun setup() {
+        themeManager = mock()
+        window = mock()
+        settings = mock()
+        whenever(window.clearFlags(anyInt())).then { }
+
+        appStore = AppStore()
+        // wait for Init action
+        appStore.waitUntilIdle()
+        binding = PrivateBrowsingBinding(
+            appStore,
+            themeManager,
+            retrieveWindow,
+            settings,
+            coroutinesTestRule.testDispatcher,
+        )
+        binding.start()
+    }
+
+    @After
+    fun teardown() {
+        binding.stop()
+    }
+
+    @Test
+    fun `WHEN mode updated THEN theme manager is also updated`() = runTest {
+        appStore.dispatch(AppAction.ModeChange(BrowsingMode.Private)).joinBlocking()
+
+        verify(themeManager).currentTheme = BrowsingMode.Private
+    }
+
+    @Test
+    fun `GIVEN screenshots not allowed in private mode WHEN mode changes to private THEN secure flag added to window`() = runTestOnMain {
+        whenever(window.addFlags(anyInt())).then { }
+        whenever(settings.allowScreenshotsInPrivateMode).thenReturn(false)
+
+        appStore.dispatch(AppAction.ModeChange(BrowsingMode.Private)).joinBlocking()
+//        this.runCurrent()
+//        this.advanceUntilIdle()
```

Comment:
Just a friendly reminder to clean up these commented lines up depending on what is needed at the end.

Reply:
Whoops, thanks 😅